NullPointerException in polyglot-translate-plugin (pom.scala)

A simple Maven project pom created from the archetype `org.apache.maven.archetypes:maven-archetype-quickstart:1.1` cannot be translated.

The `pom.xml` looks like:

``` xml
<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
  xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
  <modelVersion>4.0.0</modelVersion>

  <groupId>g</groupId>
  <artifactId>a</artifactId>
  <version>1</version>
  <packaging>jar</packaging>

  <name>a</name>
  <url>http://maven.apache.org</url>

  <properties>
    <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
  </properties>

  <dependencies>
    <dependency>
      <groupId>junit</groupId>
      <artifactId>junit</artifactId>
      <version>3.8.1</version>
      <scope>test</scope>
    </dependency>
  </dependencies>
</project>
```

And the exception is:

```
$ mvn io.takari.polyglot:polyglot-translate-plugin:0.1.9:translate -Dinput=pom.xml -Doutput=pom.scala -X
Apache Maven 3.3.3 (7994120775791599e205a5524ec3e0dfe41d4a06; 2015-04-22T13:57:37+02:00)
Maven home: /usr/share/maven-bin-3.3
Java version: 1.8.0_45, vendor: Oracle Corporation
Java home: /opt/oracle-jdk-bin-1.8.0.45/jre
Default locale: de_DE, platform encoding: UTF-8
OS name: "linux", version: "3.18.11-gentoo", arch: "amd64", family: "unix"
...
[DEBUG] Configuring mojo io.takari.polyglot:polyglot-translate-plugin:0.1.9:translate from plugin realm ClassRealm[plugin>io.takari.polyglot:polyglot-translate-plugin:0.1.9, parent: sun.misc.Launcher$AppClassLoader@4e25154f]
[DEBUG] Configuring mojo 'io.takari.polyglot:polyglot-translate-plugin:0.1.9:translate' with basic configurator -->
[DEBUG]   (f) input = /tmp/exec/a/pom.xml
[DEBUG]   (f) output = /tmp/exec/a/pom.scala
[DEBUG] -- end configuration --
[INFO] Translating /tmp/exec/a/pom.xml -> /tmp/exec/a/pom.scala
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 0.931 s
[INFO] Finished at: 2015-06-17T21:06:46+02:00
[INFO] Final Memory: 9M/212M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.takari.polyglot:polyglot-translate-plugin:0.1.9:translate (default-cli) on project a: Execution default-cli of goal io.takari.polyglot:polyglot-translate-plugin:0.1.9:translate failed. NullPointerException -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal io.takari.polyglot:polyglot-translate-plugin:0.1.9:translate (default-cli) on project a: Execution default-cli of goal io.takari.polyglot:polyglot-translate-plugin:0.1.9:translate failed.
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
        at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
        at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
        at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:307)
        at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:193)
        at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:106)
        at org.apache.maven.cli.MavenCli.execute(MavenCli.java:862)
        at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:286)
        at org.apache.maven.cli.MavenCli.main(MavenCli.java:197)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:497)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
        at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
        at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
        at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal io.takari.polyglot:polyglot-translate-plugin:0.1.9:translate failed.
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:145)
        at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
        ... 20 more
Caused by: java.lang.NullPointerException
        at org.kiama.output.PrettyPrinter$$anonfun$text$1.apply(PrettyPrinter.scala:1004)
        at org.kiama.output.PrettyPrinter$$anonfun$text$1.apply(PrettyPrinter.scala:1003)
        at org.kiama.output.PrettyPrinter$Doc.apply(PrettyPrinter.scala:978)
        at org.kiama.output.PrettyPrinter$Doc$$anonfun$$less$greater$1$$anonfun$apply$31$$anonfun$apply$32.apply(PrettyPrinter.scala:988)
        at org.kiama.output.PrettyPrinter$Doc$$anonfun$$less$greater$1$$anonfun$apply$31$$anonfun$apply$32.apply(PrettyPrinter.scala:986)
        at org.kiama.util.Trampolines$Trampoline$$anonfun$resume$1.apply(Trampolines.scala:59)
        at org.kiama.util.Trampolines$Trampoline$$anonfun$resume$1.apply(Trampolines.scala:59)
        at org.kiama.util.Trampolines$Trampoline.runT(Trampolines.scala:47)
        at org.kiama.output.PrettyPrinter$class.pretty(PrettyPrinter.scala:1112)
        at org.sonatype.maven.polyglot.scala.ScalaPrettyPrinter$.pretty(ScalaModelWriter.scala:23)
        at org.sonatype.maven.polyglot.scala.ScalaModelWriter.write(ScalaModelWriter.scala:296)
        at org.sonatype.maven.polyglot.io.ModelWriterSupport.write(ModelWriterSupport.java:47)
        at org.sonatype.maven.polyglot.TeslaModelTranslator.translate(TeslaModelTranslator.java:116)
        at org.sonatype.maven.polyglot.TeslaModelTranslator.translate(TeslaModelTranslator.java:99)
        at org.sonatype.maven.polyglot.TeslaModelTranslator.translate(TeslaModelTranslator.java:83)
        at io.takari.maven.polyglot.TranslatorMojo.translate(TranslatorMojo.java:54)
        at io.takari.maven.polyglot.TranslatorMojo.translate(TranslatorMojo.java:50)
        at io.takari.maven.polyglot.TranslatorMojo.execute(TranslatorMojo.java:43)
        at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:134)
        ... 21 more
```

--- a/polyglot-scala/src/main/scala/org/sonatype/maven/polyglot/scala/ScalaModelWriter.scala
+++ b/polyglot-scala/src/main/scala/org/sonatype/maven/polyglot/scala/ScalaModelWriter.scala
@@ -288,6 +288,13 @@ class ScalaModelWriter extends ModelWriterSupport {
     import MavenConverters._
     import ScalaPrettyPrinter._
 
+    if (mm.getModelEncoding() == null) {
+      // A null-value might be the result of parsing the model from a file with a missing '<?xml version="1.0" encoding="UTF-8"?>' header
+      // But in this case we will trigger a NPE inside of PrettyPrinter, which is really bad
+      // Thus, we restore to non-null default value
+      mm.setModelEncoding("UTF-8")
+    }
+
     val d = "import" <+> "org.sonatype.maven.polyglot.scala.model._" <@>
       "import" <+> "scala.collection.immutable.Seq" <@>
       empty <@>

